tools/shared/src/main/scala/org/scalajs/core/tools/sourcemap/SourceMapWriter.scala

```diff
@@ -50,8 +53,48 @@ class SourceMapWriter(
   private val _nameToIndex = new HashMap[String, Int]
 
   // Strings are nullable in this stack
-  private val nodePosStack = new Stack[(Position, String)]
-  nodePosStack.push((NoPosition, null))
+  private val nodePosStack = new NodePosStack
+  nodePosStack.push(NoPosition, null)
+
+  private final class NodePosStack {
+    private var topIndex: Int = -1
+    private var posStack: Array[Position] = new Array(32)
+    private var nameStack: Array[String] = new Array(32)
```

Comment:
32 doesn't seem very much. I think we can be more generous, here, and reduce the likelihood that copying is ever necessary.

What's the biggest index that is necessary when linking the test suite?